circuits/cpp/barretenberg/cpp/src/barretenberg/proof_system/circuit_builder/standard_circuit_builder.cpp

```diff
@@ -240,6 +240,237 @@ template <typename FF> void StandardCircuitBuilder_<FF>::create_poly_gate(const
     ++this->num_gates;
 }
 
+/**
+ * @brief Create a gate where we validate an elliptic curve point addition
+ *        (x1, y1) + (x2, y2) = (x3, y3)
+ *        N.B. uses incomplete addition formula. Use with caution
+ * @tparam FF
+ * @param in
+ */
+template <typename FF> void StandardCircuitBuilder_<FF>::create_ecc_add_gate(const ecc_add_gate_<FF>& in)
+{
+    const auto sign_coefficient = in.sign_coefficient;
+    const auto x1 = this->get_variable(in.x1);
+    const auto x2 = this->get_variable(in.x2);
+    const auto x3 = this->get_variable(in.x3);
+    const auto y1 = this->get_variable(in.y1);
+    const auto y2 = sign_coefficient * this->get_variable(in.y2);
+
+    bool collision = x2 == x1;
```

Comment:
why not have an assertion to ensure that the circuit fails if there is a collision?